rasa/shared/data.py

```diff
@@ -147,7 +147,7 @@ def get_data_files(
 
         if is_valid_filetype(path):
             if filter_predicate(path):
-                data_files.add(os.path.abspath(path))
+                data_files.add(path)
```

Comment:
[Rasa X ](https://github.com/RasaHQ/rasa-x/blob/b03fc7359e29a3c088a92b3dc91a8449c9216bdc/src/community/rasax/community/initialise.py#L357) also this so I'm a bit hesitant to change this as it might break things on Rasa X side. This leaves us with two options:
* changing the test: One could argue that the absolute path is bad UX 🤔 However, I think that's pretty minor
* changing [this](https://github.com/RasaHQ/rasa/blob/954e4734d479924df7211cd3bc5503285d54a66b/rasa/shared/core/trackers.py#L667) to convert the path to a relative path. 

What do you think? Both options are valid for me.

Reply:
That is so true (about the use from Rasa X). I missed it this time but next time I will be more careful on this. 
So, I changed the `test` (by changing to an absolute path) since I find this the safest solution.
I also did a search on what is the best practice (absolute vs relative path) but there are different opinions based on the use so I can't really express an opinion on that.

Reply:
Also, I added a changelog with type `improvement`.

Reply:
> I missed it this time but next time I will be more careful on this.

No worries. These things are very hard to catch and that's what a reviewer is there for. 

> I find this the safest solution.

Why do you consider this option safer?

Reply:
I think that a change in the test has less **direct** impact on the functionality of the code/product. Changing code affects directly how the code works and it can change/break things like you mentioned. However the test can also have impact on the functionality (maybe more **indirect** or **long term**?) in case the change makes us miss things that we should catch or vice versa. 🧐 Let's say that I am more of a fan of first changing/doing more tests in order to analyse the behaviour of the code under different cases/perspectives, understand better the problem it is trying to solve until I am convinced that the design of the code needs improvement or fixing. That way, taking into account different cases and having better understanding, I can also come up with a better code solution/improvement/fix proposal.

Reply:
> I think that a change in the test has less direct impact on the functionality of the code/product.

The question is **why** you had to change the test. In your case you had to change the test because you changed some production code. I've a feeling that absolute paths are in general more robust than relatives which could be seen as argument that this way is slightly "safer"